scripts/background.js

```diff
@@ -470,6 +475,72 @@ chrome.webRequest.onErrorOccurred.addListener(function(details) {
         });
       }
     }, {urls: ["<all_urls>"], types: ["main_frame"]});
+var contextMenuItemFirst={
+    "id":"first",
+    "title":"First Version",
+    "contexts":["all"]
+};
+
+var contextMenuItemRecent={
+    "id":"recent",
+    "title":"Recent Version",
+    "contexts":["all"]
+};
+var contextMenuItemAll={
+    "id":"all",
+    "title":"All Versions",
+    "contexts":["all"]
+};
+
+var contextMenuItemSave={
+    "id":"save",
+    "title":"Save Page Now",
+    "contexts":["all"]
+};
+chrome.contextMenus.create(contextMenuItemFirst);
+chrome.contextMenus.create(contextMenuItemRecent);
+chrome.contextMenus.create(contextMenuItemAll);
+chrome.contextMenus.create(contextMenuItemSave);
+chrome.contextMenus.onClicked.addListener(function(clickedData){
+    if(clickedData.menuItemId=='first'){
+        chrome.tabs.query({active: true, currentWindow: true}, function(tabs) {
+            var page_url = tabs[0].url;
+            var wayback_url ="https://web.archive.org/web/0/"
+            var pattern = /https:\/\/web\.archive\.org\/web\/(.+?)\//g;
+            var url = page_url.replace(pattern, "");
+            var open_url = wayback_url+encodeURI(url);
+            URLopener(open_url,url);
```

Comment:
There is a lot of code duplication. Please try to avoid this, lines 509 to 512 repeat 4 times!
I suggest to define a small utility function just before this point and use it 4 times.

Reply:
Why do we open the target page in the same tab (``URLopen``) in some cases while in other cases we open a new tab
 (``chrome.tabs.create``)?
Isn't it better to be consistent and always load content in the same tab?

Reply:
In all cases the target page is opened in a new tab. I have been inconsistent though with the usage of the function URLopener